File: 111. Minimum Depth of Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int minDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int left = minDepth(root.left);
        int right = minDepth(root.right);
        if (root.left == null && root.right == null) {
            return 1;
        } else if (root.left == null) {
            return right + 1;
        } else if (root.right == null) {
            return left + 1;
        } else {
            return Math.min(left, right) + 1;
        }
    }
}

File: 111. Minimum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        elif not root.left and not root.right:
            return 1
        else:
            left = self.minDepth(root.left) + 1
            right = self.minDepth(root.right) + 1
            
            if not root.left and root.right:
                return right
            elif not root.right and root.left:
                return left
            else:
                return min(left, right)
